### PR TITLE
feat(Text)!: highlighted text

### PR DESCRIPTION
**Pull request by dreamwasp**
### Overview

Deprecates the `HighlightedText` component + replaces it with a ColorMode compatible `highlight` prop on Text

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: web-1887
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change
- [x] This PR includes testing instructions tests for the code change
- [x] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### Testing Instructions

Check out the new additions to the Text story

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/31725)  | [Monolith Env](http://www.google.fr/ 'Named link title') |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/portal-app/pull/2285)  | [Portal Env](https://tengu.codecademy.com/catalog?PR_ENV=portal-app-pr-2285)   |
| Another Repo | [Another Link](http://www.google.fr/ 'Named link title') | [Another Env](http://www.google.fr/ 'Named link title')  |

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@7.4.3-alpha.8fc135.0<br/>@codecademy/gamut-illustrations@0.28.3-alpha.8fc135.0<br/>@codecademy/gamut-kit@0.6.165-alpha.8fc135.0<br/>@codecademy/gamut-labs@33.0.1-alpha.8fc135.0<br/>@codecademy/gamut-patterns@0.8.9-alpha.8fc135.0<br/>@codecademy/gamut-styles@13.4.1-alpha.8fc135.0<br/>@codecademy/gamut-tests@4.0.8-alpha.8fc135.0<br/>@codecademy/gamut@48.3.1-alpha.8fc135.0<br/>@codecademy/markdown-overrides@0.9.6-alpha.8fc135.0<br/>@codecademy/monaco@0.6.60-alpha.8fc135.0<br/>@codecademy/styleguide@59.0.1-alpha.8fc135.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62db126b3e02380ce294738c--gamut-preview.netlify.app>https://62db126b3e02380ce294738c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62db126b3e02380ce294738c>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@7.4.3-alpha.8ab617.0<br/>@codecademy/gamut-illustrations@0.28.3-alpha.8ab617.0<br/>@codecademy/gamut-kit@0.6.165-alpha.8ab617.0<br/>@codecademy/gamut-labs@33.0.1-alpha.8ab617.0<br/>@codecademy/gamut-patterns@0.8.9-alpha.8ab617.0<br/>@codecademy/gamut-styles@13.4.1-alpha.8ab617.0<br/>@codecademy/gamut-tests@4.0.8-alpha.8ab617.0<br/>@codecademy/gamut@48.3.1-alpha.8ab617.0<br/>@codecademy/markdown-overrides@0.9.6-alpha.8ab617.0<br/>@codecademy/monaco@0.6.60-alpha.8ab617.0<br/>@codecademy/styleguide@59.0.1-alpha.8ab617.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62db18b61cba530b435fb9e8--gamut-preview.netlify.app>https://62db18b61cba530b435fb9e8--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62db18b61cba530b435fb9e8>Deploy Logs</a></p>

**Review by jakemhiller — APPROVED** [review]
👍

**Review by aresnik11 — APPROVED** [review]
Love this!!

File: packages/styleguide/stories/Typography/Text.stories.mdx [hunk 1]
```diff
@@ -191,6 +191,26 @@ Text has some special props with common use cases to make your life easier.
   </LayoutGrid>
 </Story>
 
+### Highlight
+
+<Story name="Highlight">
+  <LayoutGrid gap={32} mt={24} mb={48}>
+    {typographyUtilities.screenreader.map((variant) => (
```
[the remark this answers is not in the document]
Comment (dreamwasp) [reply packages/styleguide/stories/Typography/Text.stories.mdx:198]: nah prob not, bad copy-pasta, will refactor!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@7.4.3-alpha.7ccb55.0<br/>@codecademy/gamut-illustrations@0.28.3-alpha.7ccb55.0<br/>@codecademy/gamut-kit@0.6.165-alpha.7ccb55.0<br/>@codecademy/gamut-labs@33.0.1-alpha.7ccb55.0<br/>@codecademy/gamut-patterns@0.8.9-alpha.7ccb55.0<br/>@codecademy/gamut-styles@13.4.1-alpha.7ccb55.0<br/>@codecademy/gamut-tests@4.0.8-alpha.7ccb55.0<br/>@codecademy/gamut@48.3.1-alpha.7ccb55.0<br/>@codecademy/markdown-overrides@0.9.6-alpha.7ccb55.0<br/>@codecademy/monaco@0.6.60-alpha.7ccb55.0<br/>@codecademy/styleguide@59.0.1-alpha.7ccb55.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@7.4.3-alpha.7eb484.0<br/>@codecademy/gamut-illustrations@0.28.3-alpha.7eb484.0<br/>@codecademy/gamut-kit@0.6.165-alpha.7eb484.0<br/>@codecademy/gamut-labs@33.0.1-alpha.7eb484.0<br/>@codecademy/gamut-patterns@0.8.9-alpha.7eb484.0<br/>@codecademy/gamut-styles@13.4.1-alpha.7eb484.0<br/>@codecademy/gamut-tests@4.0.8-alpha.7eb484.0<br/>@codecademy/gamut@48.3.1-alpha.7eb484.0<br/>@codecademy/markdown-overrides@0.9.6-alpha.7eb484.0<br/>@codecademy/monaco@0.6.60-alpha.7eb484.0<br/>@codecademy/styleguide@59.0.1-alpha.7eb484.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62e1432c491b0728fd691460--gamut-preview.netlify.app>https://62e1432c491b0728fd691460--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62e1432c491b0728fd691460>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@7.4.3-alpha.8a6a7b.0<br/>@codecademy/gamut-illustrations@0.28.3-alpha.8a6a7b.0<br/>@codecademy/gamut-kit@0.6.166-alpha.8a6a7b.0<br/>@codecademy/gamut-labs@33.0.2-alpha.8a6a7b.0<br/>@codecademy/gamut-patterns@0.8.9-alpha.8a6a7b.0<br/>@codecademy/gamut-styles@13.4.1-alpha.8a6a7b.0<br/>@codecademy/gamut-tests@4.0.8-alpha.8a6a7b.0<br/>@codecademy/gamut@48.4.1-alpha.8a6a7b.0<br/>@codecademy/markdown-overrides@0.9.6-alpha.8a6a7b.0<br/>@codecademy/monaco@0.6.61-alpha.8a6a7b.0<br/>@codecademy/styleguide@59.1.1-alpha.8a6a7b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://62e1a0a471e10328faa71084--gamut-preview.netlify.app>https://62e1a0a471e10328faa71084--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/62e1a0a471e10328faa71084>Deploy Logs</a></p>

**Pull request merged**